### PR TITLE
fix: App Install Dropdown Sort Properly [CAL-2285]

### LINKED ISSUE #10542: [CAL-2285] Orgs: Apps - Install app on dropdown has unusual order and "Nameless"\
* Returns "Nameless" user for me
* The order currently is
  * User
  * Team
  * Org
  * Team
  * Team

**Ultimately we want org wide apps.**

**But in the mean time we should at least do**

* User
* Org
* Team
* Team

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2285](https://linear.app/calcom/issue/CAL-2285/orgs-apps-install-app-on-dropdown-has-unusual-order-and-nameless\)</sub>

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR fixes the app install dropdown menu to show data in the correct order of:
- User
- Org
- Teams

This PR also fixes the placeholder for users that have not completed onboarding.

CleanShot 2023-08-09 at 13 31 15

Fixes #10542

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Create an org with multiple teams
- Do not complete onboarding for the org admin or delete the name in the DB
- Go to the app install page
- It should look similar to the screenshot attached

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Udit-takkar — DISMISSED** [review]
LGTM. works fine

File: packages/features/ee/teams/lib/getUserAdminTeams.ts [hunk 1]
```diff
@@ -76,7 +79,7 @@ const getUserAdminTeams = async ({
     if (user) {
       const userObject = {
         id: user.id,
-        name: user.name || "Nameless",
+        name: user.name || "Me",
```
Comment (Udit-takkar) [inline packages/features/ee/teams/lib/getUserAdminTeams.ts:82]: NIT: use i18n here

File: apps/web/components/apps/InstallAppButtonChild.tsx [hunk 2]
```diff
@@ -112,7 +112,7 @@ export const InstallAppButtonChild = ({
                   );
                 }}>
                 <p>
-                  {team.name} {isInstalled && `(${t("installed")})`}
+                  {t(team.name)} {isInstalled && `(${t("installed")})`}
```
Comment (joeauyeung) [inline apps/web/components/apps/InstallAppButtonChild.tsx:115]: The `team.name` will cover "me" so it'll grab the translation. Any other team name will be passed through as normal. @Ud

**Review by Udit-takkar — DISMISSED** [review]

**Review by leog — APPROVED** [review]
LGTM, thanks for taking care of this

**Pull request merged**